### ISSUE TITLE
3.2.2+ Cannot update identity column 'Identity Column Id'.

### ISSUE THREAD

**Issue by jdmoore1618** [borisdj/EFCore.BulkExtensions#403]
Hello,

I upgraded to 3.2.3 today and I am experiencing an error while trying to upsert (insert) multiple entries into the database. I get the following error "Cannot update identity column 'Identity Column Id'." My identity column is defined as "Id" and is zero at the time on insertion for each record.

Table defined as follows:

```
public class TableName: IIdentifiable<int>
	{
		[Column("TableNameId")]
		public int Id { get; set; }

		....more properties
	}
```

Table is included in context as follows:

```
public partial class Context: DbContext
	{
		public DbSet<TableName> TableName { get; set; }
	}
```

We generically use BulkInsertOrUpdateAsync like so:

```
		public async Task<IEnumerable<TDomainModel>> BulkUpsertAsync<TDomainModel>(IEnumerable<TDomainModel> domainModels, CancellationToken cancellationToken)
			where TDomainModel : class, IIdentifiable<int>
		{
			var domainModelsToList = domainModels.ToList();

			await _context.BulkInsertOrUpdateAsync(
				domainModelsToList,
				new BulkConfig { SetOutputIdentity = true,  },
				null,
				cancellationToken);

			return domainModelsToList;
		}
```

Before updating the above code would insert records into the database just fine but now I am getting an error. Can you assist?

Thanks.

**Comment by borisdj** [comment]
Fixed with 3.2.4

**Comment by schil227** [comment]
This appears to be broken again in 3.2.7 - had the same issue, downgrading to 3.2.4 solved it.
